nvm.sh

```diff
@@ -2724,26 +2727,25 @@ nvm() {
     return
   fi
 
-  local DEFAULT_IFS
-  DEFAULT_IFS=" $(nvm_echo t | command tr t \\t)
+  local IFS
+  IFS=" $(nvm_echo t | command tr t \\t)
 "
-  if [ "${-#*e}" != "$-" ]; then
-    set +e
-    local EXIT_CODE
-    IFS="${DEFAULT_IFS}" nvm "$@"
-    EXIT_CODE="$?"
-    set -e
-    return "$EXIT_CODE"
-  elif [ "${-#*a}" != "$-" ]; then
-    set +a
+  local OPTIONS=''
+  local O='e a'
+  # set +E only disables errtrace on Bash
+  [ -z "${BASH-}" ] || O='e a E'
+  for O in $O; do
+    if [ "${-#*$O}" != "$-" ]; then
+      set "+$O"
+      OPTIONS="${OPTIONS}${O}"
+    fi
+  done
+  if [ -n "$OPTIONS" ]; then
     local EXIT_CODE
-    IFS="${DEFAULT_IFS}" nvm "$@"
+    nvm "$@"
```

Comment:
This indeed isn’t the same; non-exported variables don’t inherit to subshells everywhere.
```suggestion
    IFS=“${IFS}” nvm "$@"
```
that’s why i named it default_ifs - to highlight that it’s different on purpose.

Reply:
Ahhh. I see ksh behaves differently to the others here. I'm tempted to test something like

```sh
local IFS
unset IFS
export -n IFS
```

to achieve a cleaner result, but I think that would fall outside the scope of the PR 😉

Reply:
That would also risk changing the user's shell environment, into which nvm is sourced, so that's not an option. I prefer reverting the name and keeping the explicit invocation :-)

Reply:
Theoretically, `local IFS` should protect the user's environment (it does on Bash, at least), but it would definitely need testing given `unset` is probably implemented differently by each shell. One for a rainy day.

Reply:
ish doesn’t support local.